ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml

```diff
@@ -313,6 +323,32 @@ QObject {
                 console.error("No password or pin provided to sign message")
             }
         }
+
+        // Returns AppLayouts.Wallet.services.dapps.types->SessionRequest.EstimatedTime
+        function getEstimatedTimeInterval(data, method, chainId) {
+            if (method != SessionRequest.methods.signTransaction.name
+                && method != SessionRequest.methods.sendTransaction.name)
+            {
+                return ""
+            }
+
+            var tx = {}
+            if (method === SessionRequest.methods.signTransaction.name) {
+                tx = SessionRequest.methods.signTransaction.getTxObjFromData(data)
+            } else if (method === SessionRequest.methods.sendTransaction.name) {
+                tx = SessionRequest.methods.sendTransaction.getTxObjFromData(data)
+            }
+
+            // Empty string instructs getEstimatedTime to fetch the blockchain value
+            var maxFeePerGas = ""
+            if (!!tx.maxFeePerGas) {
+                maxFeePerGas = tx.maxFeePerGas
+            } else if (!!tx.gasPrice) {
+                maxFeePerGas = tx.gasPrice
+            }
+
+            return root.store.getEstimatedTime(chainId, maxFeePerGas)
```

Comment:
What `store` contains this? I can't find it here

Reply:
The `DAppsStore`; please note that it is added in this PR, see: https://github.com/status-im/status-desktop/blob/f74448e81ef70f61fa39432e658186ef5289e688/ui/imports/shared/stores/DAppsStore.qml#L74